substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/DebugInfoBase.java

```diff
@@ -119,36 +160,95 @@ public void installDebugInfo(DebugInfoProvider debugInfoProvider) {
          */
 
         /*
-         * Ensure we have a null string in the string section.
+         * track whether we need to use a heap base regsiter
+         */
+        useHeapBase = debugInfoProvider.useHeapBase();
+
+        /*
+         * save count of low order tag bits that may appear in references
          */
+        int oopTagsMask = debugInfoProvider.oopTagsMask();
+
+        /* tag bits must be between 1 and 32 for us to emit as DW_OP_lit<n> */
+        assert oopTagsMask > 0 && oopTagsMask < 32;
+        /* mask must be contiguous from bit 0 */
+        assert ((oopTagsMask + 1) & oopTagsMask) == 0;
+
+        oopTagsCount = Integer.bitCount(oopTagsMask);
+
+        /* Save amount we need to shift references by when loading from an object field. */
+        oopCompressShift = debugInfoProvider.oopCompressShift();
+
+        /* shift bit count must be either 0 or 3 */
+        assert (oopCompressShift == 0 || oopCompressShift == 3);
+
+        /* Save number of bytes in a reference field. */
+        oopReferenceSize = debugInfoProvider.oopReferenceSize();
+
+        /* Save alignment of a reference */
+        oopAlignment = debugInfoProvider.oopAlignment();
+
+        /* Save alignment of a reference */
+        oopAlignShift = Integer.bitCount(oopAlignment - 1);
+
+        /* reference alignment must be 8 bytes */
```

Comment:
Sometimes you comments start with lowercase ...

Reply:
Yeah, sorry for that. I keep forgetting about this. I think I have found all such cases and corrected them.